pkg/autoscaler/multiscaler_test.go

```diff
@@ -160,6 +164,34 @@ func TestMultiScalerScaleToZero(t *testing.T) {
 		t.Fatalf("timed out waiting for Watch()")
 	}
 
+	// Verify that we stop seeing "ticks"
+	select {
+	case <-done:
+		t.Fatalf("Got unexpected tick")
+	case <-time.After(30 * time.Millisecond):
+		// We got nothing!
+	}
+
+	desiredScaleMutex.Lock()
+	desiredScale = 1
+	desiredScaleMutex.Unlock()
+	uniScaler.setScaleResult(1, true)
+	now := time.Now()
+	testStat := autoscaler.Stat{
+		Time:                      &now,
+		PodName:                   "test-pod",
+		AverageConcurrentRequests: 3.5,
+		RequestCount:              20,
+	}
+	ms.RecordStat(testKPAKey, testStat)
+	// Verify that we see a "tick" on scale from zero
+	select {
+	case <-done:
+		// We got the signal!
+	case <-time.After(30 * time.Millisecond):
+		t.Fatalf("timed out waiting for Watch()")
+	}
```

Comment:
I think this test would also pass without the changes made? I get coverage on the newly added path only intermittently. The multiscaler runs on a tick interval of 1ms so it will be quick either way. Should we increase the tick interval of the multiscaler and assert we get a tick in a substantially quicker time? (For example: tickInterval of 100ms, make the others wait for 200ms and this wait for 30ms).

Reply:
+1 let's add a parallel test to this with a tick interval larger than the overall test timeout.  Then we should be able to properly test this, right?

Reply:
Hrm, I am having a hard time coming up with a way to do this without chopping up of adding some multiscaler access. When we create a metric which in turn creates a scaler we need to go through one tickScaler cycle in order for the scaler desiredScale to be set to 0. If I up the tick interval then we'll have to wait for that time period. AFAICT theres no good interface for me to inject this tick cycle, do you have any thoughts on this @markusthoemmes ?

Reply:
@markusthoemmes hrm, check out what I just pushed - I added a method to directly set the desiredscale of a scalerrunner which lets us actually test this code path 100% of the time.